src/wp-includes/media.php

```diff
@@ -1958,6 +1971,104 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 	return $image;
 }
 
+/**
+ * Adds optimization attributes to an `img` HTML tag.
+ *
+ * @since 6.3.0
+ *
+ * @param string $image   The HTML `img` tag where the attribute should be added.
+ * @param string $context Additional context to pass to the filters.
+ *
+ * @return string Converted `img` tag with optimization attributes added.
+ */
+function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
+	$width              = preg_match( '/ width="([0-9]+)"/', $image, $match_width ) ? (int) $match_width[1] : '';
+	$height             = preg_match( '/ height="([0-9]+)"/', $image, $match_height ) ? (int) $match_height[1] : '';
+	$optimization_attrs = wp_get_loading_optimization_attributes(
+		'img',
+		array(
+			'width'  => $width,
+			'height' => $height,
+		),
+		$context
+	);
+
+	/**
+	 * Filters the loading optimization attributes to be added to an image.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string|bool $optimization_attrs   Attribute and it's values obtained from `wp_get_loading_optimization_attributes` function.
+	 * @param string      $image   The HTML `img` tag to be filtered.
+	 * @param string      $context Additional context about how the function was called or where the img tag is.
+	 */
+	$optimization_attrs = apply_filters( 'wp_img_tag_add_loading_optimization_attrs', $optimization_attrs, $image, $context );
```

Comment:
@felixarntz @joemcgill  I have introduced a new filter so user can modify the final values. ( inspired from `wp_img_tag_add_loading_attr` filter.

Reply:
Removed the filter.